### PR TITLE
AvroProducer: fetch schema to cache by id during produce

### PR DESCRIPTION
**Pull request by mfe5003**
This pull request allows an `AvroProducer` to fetch a schema from the schema registry by schema id, if missing from the local registry cache.
For issue #248

### CODE REVIEW

**Comment by mfe5003** [comment]
[clabot:check]

File: confluent_kafka/avro/__init__.py [hunk 1]
```diff
@@ -26,7 +26,10 @@ class AvroProducer(Producer):
     """
 
     def __init__(self, config, default_key_schema=None,
-                 default_value_schema=None, schema_registry=None):
+                 default_value_schema=None,
```
Comment (ewencp) [inline confluent_kafka/avro/__init__.py]: may not be ideal, but it's better to not rearrange the order of existing arguments since users could already be calling this without using keyword arguments

**Comment by mfe5003** [comment]
I am migrating an existing IoT system that uses a format very similar to Avro + schema-registry.
The device registers a schema and receives unique identifier which it prepends to all outgoing binary data.
My producers then need to do the translation from the old format to the new Avro before appending to the topic, and don't necessarily have the schema cached, so they may need to check the registry.

Basically [this situation](https://gist.github.com/mfe5003/16140f8255f2a9472e1e762d49b9ee15) is what I am looking for, where the schema_id would be known to a client that interacts with the producers.

**Comment by mfe5003** [comment]
I have overlooked the fact that for my application I need to look up the topic by the associated schema id.  This implicitly assumes that the schema id and topic have a 1-to-1 mapping which is true for my application, but not in general.  I am closing this PR, sorry for the confusion.
